server/core/static/external/cred_update.js

```diff
@@ -0,0 +1,58 @@
+function setupInteractivePwdModalListeners() {
+    console.log("rehooked input listeners")
+    const new_pwd = document.getElementById("new-password");
+    const new_pwd_check = document.getElementById("new-password-check");
+    const pwd_submit = document.getElementById("password-submit");
+
+    function markPwdCheckValid() {
+        new_pwd_check.classList.remove("is-invalid");
+        new_pwd_check.classList.add("is-valid");
+        pwd_submit.disabled = false;
+    }
+
+    function markPwdCheckInvalid() {
+        new_pwd_check.classList.add("is-invalid");
+        new_pwd_check.classList.remove("is-valid");
+        pwd_submit.disabled = true;
+    }
+
+    new_pwd.addEventListener("input", (_) => {
+        if (new_pwd.value === new_pwd_check.value) {
```

Comment:
Minor UX thing, it'd be nice to differentiate between "You haven't put anything in the second box" and "the two are different"... I always find it annoying when I'm getting red-flagged as "You're doing it wrong" before I've even started typing in the second field 😞 We still need to note that the second box needs filling, but ... not THEY ARE DIFFERENT 😄

Reply:
Yeah that's true, I'll change that :)